File: persistenece/src/main/java/com/anokmik/persistence/model/PhotoAttachment.java
package com.anokmik.persistence.model;

import com.anokmik.persistence.TripAssistantDatabase;
import com.raizlabs.android.dbflow.annotation.Column;
import com.raizlabs.android.dbflow.annotation.ForeignKey;
import com.raizlabs.android.dbflow.annotation.ForeignKeyReference;
import com.raizlabs.android.dbflow.annotation.PrimaryKey;
import com.raizlabs.android.dbflow.annotation.Table;
import com.raizlabs.android.dbflow.structure.BaseModel;

@Table(database = TripAssistantDatabase.class)
public final class PhotoAttachment extends BaseModel {

    @PrimaryKey(autoincrement = true)
    public long id;

    @ForeignKey(references = {
            @ForeignKeyReference(
                    columnType = long.class,
                    columnName = "tripEvent",
                    foreignKeyColumnName = "id"
            )
    })
    @Column
    public TripEvent tripEvent;

    @Column
    public String path;

}

File: persistenece/src/main/kotlin/com/anokmik/persistence/model/PhotoAttachment.kt
package com.anokmik.persistence.model

import com.anokmik.persistence.TripAssistantDatabase
import com.raizlabs.android.dbflow.annotation.*
import com.raizlabs.android.dbflow.structure.BaseModel

@Table(database = TripAssistantDatabase::class)
class PhotoAttachment : BaseModel() {

    @JvmField
    @PrimaryKey(autoincrement = true)
    var id: Long = 0

    @JvmField
    @ForeignKey(references = arrayOf(
            ForeignKeyReference(
                    columnType = Long::class,
                    columnName = "tripEvent",
                    foreignKeyColumnName = "id"
            )
    ))
    @Column
    var tripEvent: TripEvent? = null

    @JvmField
    @Column
    var description: String? = null

    @JvmField
    @Column
    var path: String? = null

}